framework/src/mesh/FileMesh.C

```diff
@@ -74,6 +82,30 @@ FileMesh::buildMesh()
     getMesh().prepare_for_use();
     getMesh().skip_partitioning(skip_partitioning_later);
   }
+  else if (_is_checkpoint)
+  {
+    CheckpointIO reader(getMesh());
+
+    reader.binary() = true;
+
+    auto & mesh = getMesh();
+
+    DistributedMesh * pmesh = cast_ptr<DistributedMesh *>(&mesh);
+
+    if (pmesh)
+      reader.parallel() = true;
```

Comment:
Actually, this whole case should be unnecessary now.  The current libMesh CheckpointIO class determines whether it's *input* is parallel, regardless of it's own setting, and it should handle reading serial files into a parallel mesh or vice-versa.  The parallel() option only affects *output*.